### PR TITLE
Fix Rest API cache-control response header 

### LINKED ISSUE #7486: Rest API incorrect cache control configuration for some endpoints
### Description

We have custom cache-control for several rest api endpoints in application.yml, however, there are two issues:

- the path pattern for transaction by id or hash has changed from `/api/v1/transactions/:transactionId` to `/api/v1/transactioins/:transactionIdOrHash`
- the actual path pattern for `/api/v1/network/exchangerate` and `/api/v1/network/fees` in req.route.path are `/exchangerate` and `/fees` respectively

### Steps to reproduce

run the rest server locally, check for the `cache-control` header in response for different endpoints, compare it against to what's set in application.yml

### Additional context

_No response_

### Hedera network

other

### Version

v0.97.0-SNAPSHOT

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR fixes the incorrect cache-control response header configuration

- Fix cache-control path pattern for transaction by id or hash endpoint
- Fix cache-control path pattern for network exchangerate and fees endpoints
- Change swagger metrics authentication default to false and remove override in helm charts

**Related issue(s)**:

Fixes #7486 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/config/application.yml [hunk 1]
```diff
@@ -67,15 +67,14 @@ hedera:
       response:
         compression: true
         headers:
-          default: { "Cache-Control": "public, max-age=1" }
+          default: { "cache-control": "public, max-age=1" }
```
Comment (xin-hedera) [inline hedera-mirror-rest/config/application.yml:70]: change to lower case to match what's in express response

File: hedera-mirror-rest/config/application.yml [hunk 2]
```diff
@@ -67,15 +67,14 @@ hedera:
...
-              { "Cache-Control": "public, max-age=60" }
-            /api/v1/transactions/:transactionId:
-              { "Cache-Control": "public, max-age=300" }
+            /api/v1/transactions/:transactionIdOrHash:
+              { "cache-control": "public, max-age=300" }
             /api/v1/transactions/:transactionId/stateproof:
-              { "Cache-Control": "public, max-age=600" }
+              { "cache-control": "public, max-age=600" }
+            /exchangerate: { "cache-control": "public, max-age=60" }
+            /fees: { "cache-control": "public, max-age=60" }
```
Comment (xin-hedera) [inline hedera-mirror-rest/config/application.yml:77]: for handlers mounted under a Router, `req.route.path` only contains the part in the scope of the router. For example, the router is mounted on to `/trunk`, and a handler then is mounted at `/branch` onto the router, inside the handler, `req.route.path` is just `/branch`.

Per expressjs doc, `req.baseUrl` is supposed to be `/trunk`, however it looks like awaitjs/express breaks it.

As a result, the PR resorts to use the path inside the router for exchangerate and fees now.

**Review by edwin-greene — APPROVED** [review]
Looks good

File: hedera-mirror-rest/config/application.yml (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-rest/config/application.yml:77]: turns out it's not awaitjs/express.  `req.baseUrl` being empty in the responseHandler is expected. Create a follow-up ticket here: https://github.com/hashgraph/hedera-mirror-node/issues/7489

**Pull request merged**